orderbook/src/price_estimate.rs

```diff
@@ -44,6 +46,7 @@ impl UniswapPriceEstimator {
 impl PriceEstimating for UniswapPriceEstimator {
     // Estimates the price using the direct pool between sell and buy token.
     // Returns an error if no pool exists between sell and buy token.
+    // Incorporates uniswap fee unless amount is 0 in which case it returns the best spot price.
```

Comment:
For the spot price it looks like the OrderKind does not matter. Maybe it would make sense to make this a separate function `estimate_spot_price` in that case? I'm not sure, just something I noticed when reading.

Reply:
We could do that. What should then be the behavior of estimate price when invoked with 0 amount? Would it call into that method (in this case it would just be an additional method on PriceEstimating) or should it returns an error? I feel the rounding to 1 we do right now will lead to very bad results.

Reply:
I would say that you are not allowed to call estimate_price with 0 amount because it is not a meaningful value for this function.

Reply:
Right, I'm slightly worried that in this case the caller will have to always do a if amount == 0 estimate_price else estimate spot_price (since e.g. the amount for gas estimation can be 0 if the gas price is 0, or U256 user input can always be 0.

That's why I'm slightly favouring handling it inside the estimate_price abstraction directly (although I see your point that it's not great to implicitly apply fundamentally different logic for slightly different values).

Reply:
That's reasonable too. I'm fine with either decision. Mostly wanted to point out the idea in case you hadn't thought about it.